pkg/cover/html.go

```diff
@@ -26,22 +26,23 @@ import (
 
 func fixUpPCs(target string, progs []Prog, coverFilter map[uint32]uint32) []Prog {
 	if coverFilter != nil {
-		for _, prog := range progs {
+		for i, prog := range progs {
 			var nPCs []uint64
 			for _, pc := range prog.PCs {
 				if coverFilter[uint32(pc)] != 0 {
 					nPCs = append(nPCs, pc)
 				}
 			}
 			prog.PCs = nPCs
+			progs[i] = prog
```

Comment:
I don't understand this. We just read prog from progs[i], why do we need to assign it back?

Reply:
prog.PCs = nPCs
prog has been updated with new nPCs. and then progs passed to prepareFileMap. If we don't do progs[i] = prog, the pc inside prepareFileMap is still old value.

Reply:
In the case of old value, progPCs[pc] will likely to fail.

Reply:
Ah, I see, I missed that progs are stored by value.